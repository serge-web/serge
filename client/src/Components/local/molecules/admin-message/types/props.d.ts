import { MessageChannel } from 'src/custom-types'

export default interface PropTypes {
  /**
   * The message content
   */
  message: MessageChannel
  /**
   * Current roleId
   */
  roleId?: string
  /**
   * Current forceId
   */
  force?: string
}
