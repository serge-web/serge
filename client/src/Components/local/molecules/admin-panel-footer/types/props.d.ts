import { Role } from '@serge/custom-types'
import React from 'react'

export default interface Props {
  force: {
    color: string
    name: string
    /**
    * @deprecated use iconURL instead
    */
    icon?: string
    iconURL: string
  }
  selectedRoleName: Role['name']
  byPassUrl?: string
  onIconClick: React.ReactEventHandler
}
