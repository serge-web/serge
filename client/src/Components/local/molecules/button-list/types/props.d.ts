import { Props as ButtonProps } from '../../../atoms/button/types/props'

export type Props = {
  /**
   * List of button configurations
   */
  buttons: Array<ButtonProps>
}
