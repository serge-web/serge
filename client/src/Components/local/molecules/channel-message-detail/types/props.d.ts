import { MessageCustom } from 'src/custom-types'

export default interface Props {
  /**
   * Message detail object
   * TODO: produce a TypeScript interface for the detail contents
   */
  message: MessageCustom 
  /**
   * Checks if user is an umpire. Assume not umpire if parameter missing
   */
  isUmpire: boolean
  /**
   * If control should be rendered in collapsed state
   */
  collapsed?: boolean
}
