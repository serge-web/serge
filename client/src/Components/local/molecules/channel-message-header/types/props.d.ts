import { Role } from 'src/custom-types'
import React from 'react'

export default interface Props {
  /**
   * Header collapsible open state
   */
  isOpen?: boolean
  /**
   * Header item title
   */
  title?: string
  /**
   * Message timestamp
   */
  timestamp?: string
  /**
   * Message sender role
   */
  role?: Role['roleId']
  /**
   * Message sender rolename
   */
  roleName?: Role['name']
  /**
  * Message force color
  */
  forceColor: string
  /**
   * Message type
   */
  messageType?: string
  /**
   * Message read state
   */
  hasBeenRead?: boolean
  /**
   * Handle on collapse event
   */
  onExpand?: React.ReactEventHandler

  markUnread?: () => void
}
