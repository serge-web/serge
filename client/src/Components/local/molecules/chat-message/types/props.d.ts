import { ChatMessage } from 'src/custom-types'

export type Props = {
    message: ChatMessage
    /**
     * Checks if user is from umpire force
     */
    isUmpire: boolean
    isOwner: boolean
    markUnread?: (message: ChatMessage) => void
    /** whether to hide the author (force/role) of the message */
    hideAuthor?: boolean
  }
