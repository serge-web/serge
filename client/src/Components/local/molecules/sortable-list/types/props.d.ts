import React, { ReactNode, ReactText } from 'react'

export interface ItemObject {
  name: string
  uniqid?: string | number
  [property: string]: any
}

export type Item = number | string | ItemObject

export interface SortableItem {
  /** The unique id associated with your item. It's recommended this is the same as the key prop for your list item. */
  id: ReactText
  /** When true, the item is selected using MultiDrag */
  selected?: boolean
  /** When true, the item is deemed "chosen", which basically just a mousedown event. */
  chosen?: boolean
  /** When true, it will not be possible to pick this item up in the list. */
  filtered?: boolean
  item: Item
  [property: string]: any
}

export default interface PropTypes {
  items: Array<Item>
  onClick?: (item: Item, key: number, e?: any) => void
  onChange?: (items: Array<Item>) => void
  onCreate?: () => void
  renderItemSection?: (items: Item, key: number) => ReactNode
  title?: string
  copy?: boolean
  /** whether to show remove/trash icon */
  remove: boolean
  sortable?: 'manual' | 'auto'
  required?: boolean
  valueOnEmpty?: ReactText
  /** Custom delete handler.
   * Initially used for ensuring Game Control isn't being deleted */
  customDeleteHandler?: (NewItems: Item[], key: number, handleChange: (changedItems: Array<Item>) => void) => void
  viewDirection?: 'horizontal' | 'vertical'
  disableButtonAdd?: boolean
  ref?: React.Ref<HTMLDivElement>
}
