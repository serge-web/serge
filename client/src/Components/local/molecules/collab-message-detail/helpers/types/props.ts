import { CollaborativeMessageStates } from 'src/config'
import { ForceRole } from 'src/custom-types'

interface Props {
    isResponse: boolean
    canSeeResponse: boolean
    message: any 
    collaboration: any 
    template: any 
    role: ForceRole
    state: CollaborativeMessageStates
    responseHandler: (val: { [property: string]: any }) => void
    newMessageHandler: (val: { [property: string]: any }) => void
    // formIsEditable: boolean | undefined
    gameDate: string
}

export default Props
