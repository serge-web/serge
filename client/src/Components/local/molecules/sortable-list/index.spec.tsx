/* global it expect */

import React from 'react'
import renderer from 'react-test-renderer'

import SortableList from './index'
import { watuWargame } from 'src/mocks'

import { Item } from './types/props'
const forcesMock = watuWargame.data.forces.forces[0].roles

const items: Array<Item> = forcesMock

it('sortable-list renders correctly', () => {
  const tree = renderer
    .create(<SortableList remove={true} items={items}>Text</SortableList>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
