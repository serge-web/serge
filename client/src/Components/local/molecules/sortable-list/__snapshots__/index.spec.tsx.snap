// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sortable-list renders correctly 1`] = `
<div
  className="main"
>
  <ul>
    <li
      className="li"
      data-id="0"
    >
      <div
        className="drag"
      >
        <span>
          <div
            onClick={[Function]}
          >
            <svg
              className="svg-inline--fa fa-grip-vertical"
            />
          </div>
        </span>
      </div>
      <div
        className="presection"
      >
        <section
          className="section"
        >
          <div
            className="value-label"
          >
            <div
              onClick={[Function]}
            >
              Game Control
            </div>
          </div>
          <span>
            <div
              className="trash-icon"
              onClick={[Function]}
            >
              <svg
                className="svg-inline--fa fa-trash"
              />
            </div>
          </span>
          <span />
        </section>
      </div>
    </li>
    <li
      className="li"
      data-id="1"
    >
      <div
        className="drag"
      >
        <span>
          <div
            onClick={[Function]}
          >
            <svg
              className="svg-inline--fa fa-grip-vertical"
            />
          </div>
        </span>
      </div>
      <div
        className="presection"
      >
        <section
          className="section"
        >
          <div
            className="value-label"
          >
            <div
              onClick={[Function]}
            >
              RFI Mgr
            </div>
          </div>
          <span>
            <div
              className="trash-icon"
              onClick={[Function]}
            >
              <svg
                className="svg-inline--fa fa-trash"
              />
            </div>
          </span>
          <span />
        </section>
      </div>
    </li>
    <li
      className="li"
      data-id="2"
    >
      <div
        className="drag"
      >
        <span>
          <div
            onClick={[Function]}
          >
            <svg
              className="svg-inline--fa fa-grip-vertical"
            />
          </div>
        </span>
      </div>
      <div
        className="presection"
      >
        <section
          className="section"
        >
          <div
            className="value-label"
          >
            <div
              onClick={[Function]}
            >
              Blue HHQ
            </div>
          </div>
          <span>
            <div
              className="trash-icon"
              onClick={[Function]}
            >
              <svg
                className="svg-inline--fa fa-trash"
              />
            </div>
          </span>
          <span />
        </section>
      </div>
    </li>
    <li
      className="li"
      data-id="3"
    >
      <div
        className="drag"
      >
        <span>
          <div
            onClick={[Function]}
          >
            <svg
              className="svg-inline--fa fa-grip-vertical"
            />
          </div>
        </span>
      </div>
      <div
        className="presection"
      >
        <section
          className="section"
        >
          <div
            className="value-label"
          >
            <div
              onClick={[Function]}
            >
              Red HHQ
            </div>
          </div>
          <span>
            <div
              className="trash-icon"
              onClick={[Function]}
            >
              <svg
                className="svg-inline--fa fa-trash"
              />
            </div>
          </span>
          <span />
        </section>
      </div>
    </li>
  </ul>
  <div
    className="add-section"
  >
    <button
      className="button"
      onClick={[Function]}
    >
      Add
    </button>
  </div>
</div>
`;
